starters/gatsby-wordpress-starter/__tests__/snapshotTests/postsIndex.test.jsx

```diff
@@ -1,27 +1,24 @@
-import * as Gatsby from 'gatsby'
 import { render } from '@testing-library/react'
 
 import PostIndexTemplate from '../../src/templates/postsIndex'
 import data from '../data/allWpPost.json'
 
-const StaticQuery = vi.spyOn(Gatsby, 'StaticQuery')
-
 describe('<PostIndexTemplate />', () => {
 	beforeEach(() => {
-		StaticQuery.mockImplementation(() => {
-			return {
-				data,
-			}
-		})
+		location = {
+			pathname: '/posts/1',
+			state: { breakOpen: false },
+		}
 	})
 
 	it('should render with posts', () => {
 		const { asFragment } = render(
 			<PostIndexTemplate
-				data={data}
+				location={location}
 				pageContext={{
-					nextPagePath: data.allWpPost.nodes[1].uri,
-					previousPagePath: null,
+					itemsPerPage: 2,
+					routing: true,
+					posts: [{ post: data.allWpPost.nodes[0] }],
```

Comment:
Any reason not to pass in all of the posts here?
```suggestion
					posts: data.allWpPost.nodes,
```

Reply:
No reason, and I can add the second post in the example data if we want. The reason it is like this has to do with how the test data is structured and what the component is expecting. The `<PostIndexTemplate />` and `<Post />` are now passed data in different structures.  Data for `<PostIndexTemplate />` is now coming from `gatsby-node` which has a GraphQL alias `post`, this syntax is there to replicate the structure but not change the actual test data. Since the test data is not changing the same file can be used for the `<Post />` test.

Reply:
Short answer - We can pass in a whole array of posts but the test data doesn't have the `post` alias that is expected. I could add it in the test data but then we would need two separate data files for the `<Post />` and `<PostIndexTemplate />` tests

Reply:
If the data structure is different, it should have its own json file instead of reformatting the other test data imo.

Reply:
Ah I got mixed up in writing the tests, the same dataset should be fine for each. The data they receive is in the same shape with the `post` alias, they just destructure it differently. Updated the data set with the alias but they should be good to share 👍

Reply:
Right on, thanks!